lib/hnvm/config.sh

```diff
@@ -12,7 +12,42 @@ while [ -h "$source" ]; do # resolve $source until the file is no longer a symli
 done
 script_dir="$( cd -P "$( dirname "$source" )" >/dev/null 2>&1 && pwd )"
 
-export COMMAND_OUTPUT=${HNVM_OUTPUT_DESTINATION:-/dev/stdout}
+COMMAND_OUTPUT=""
+
+# Try aggressively to find an environment-supported, redirectable, stdout fd descriptor
+#
+# In nodejs spawned subprocess (think npm pre/postinstall scripts), stdio can be
+# provided as sockets, which is not able to do simple '>>' bash redirection
+# https://nodejs.org/api/child_process.html#optionsstdio
+# "These are not actual Unix pipes and therefore the child process can not use them by their
+# descriptor files, e.g. /dev/fd/2 or /dev/stdout."
+#
+# For command output shell redirection, hierarchy is (most to least priority)
+# - $HNVM_OUTPUT_DESTINATION - arbitrary path given as env var
+# - /dev/stdout - convenience symlink to fd 1 - not fully posix portable
+# - /dev/fd/1 - on macos, the character special device (c in `ls`) that is the stdout stream
+# - /dev/null - only used as fallback, if none of the above exist
+if [[ -n "$HNVM_OUTPUT_DESTINATION" && ! -S "$HNVM_OUTPUT_DESTINATION" ]]; then
```

Comment:
Try aggressively to find an environment-supported, redirectable, stdout fd descriptor

In nodejs spawned subprocess (think npm pre/postinstall scripts), stdio can be
provided as sockets, which is not able to do simple '>>' bash redirection
https://nodejs.org/api/child_process.html#optionsstdio
> "These are not actual Unix pipes and therefore the child process can not use them by their descriptor files, e.g. /dev/fd/2 or /dev/stdout."

For command output shell redirection, hierarchy is (most to least priority)
- $HNVM_OUTPUT_DESTINATION - arbitrary path given as env var
- /dev/stdout - convenience symlink to fd 1 - not fully posix portable
- /dev/fd/1 - on macos, the character special device (c in `ls`) that is the stdout stream
- /dev/null - only used as fallback, if none of the above exist

Reply:
The main motive for this hierarchy, is to prevent this from ever happening:

- https://compass-tech.atlassian.net/browse/TECH-7725
- https://app.circleci.com/pipelines/github/UrbanCompass/urbancompass/993565/workflows/454d4c7d-e413-47da-a495-1cfd7d0836cf
```
npm ERR! path /home/myuser/myproject/node_modules/aws-sdk
npm ERR! command failed
npm ERR! command sh -c node scripts/warn-maintenance-mode.js
npm ERR! /home/myuser/.basher/cellar/packages/UrbanCompass/hnvm/lib/hnvm/config.sh: line 215: /dev/stdout: No such device or address
```